docs/SIGNAL_ROADMAP.MD

```diff
@@ -1,5 +1,56 @@
+4.  [ ] **Erweiterung für mechanische Signale:** Definition und Implementierung der ersten mechanischen Signale (z.B. deutsches `Hp`-Formsignal) unter Nutzung des Servo-Subsystems.
+5.  [ ] **Umfassendes Unit-Testing-Framework:** Aufbau einer soliden Test-Suite (z.B. mit GoogleTest) für die Kern-Logik (DCC-Parser, Signal-Dispatcher, etc.), um Stabilität zu gewährleisten.
+
+**Phase 2: Konnektivität & Benutzererfahrung (UX)**
+6.  [ ] **Integriertes Web-Interface:** Ein auf dem Decoder laufender Webserver zur grafischen Konfiguration aller CVs und Decoder-Optionen über den Browser.
+7.  [ ] **WLAN-Konnektivität (On-Board):** Nutzung eines Mikrocontrollers mit WLAN (z.B. ESP32) oder eines Co-Prozessors, um das Web-Interface bereitzustellen.
+8.  [ ] **RailCom-Implementierung (Cutout & Channel 1):** Basis-Implementierung von RailCom zur Übermittlung der Decoder-Adresse und für PoM-CV-Lesen.
+9.  [ ] **Live-Vorschau im Web-Interface:** Grafische Darstellung des konfigurierten Signals und seiner Aspekte direkt im Browser, analog zum bestehenden Web-Tool.
+10. [ ] **Firmware-Updates über Web-Interface (OTA):** Ermöglichen von "Over-the-Air"-Firmware-Updates, ohne dass der Decoder ausgebaut werden muss.
+
+**Phase 3: Funktionserweiterungen & Peripherie**
+11. [ ] **Weichen-Decoder-Modus:** Implementierung eines Modus, um den Decoder als Weichendecoder für Spulen- oder Servo-Antriebe zu verwenden.
+12. [ ] **Blockstellen-Logik-Modul:** Integrierte, einfache Blockstellen-Logik, die über CVs konfiguriert werden kann, um Abhängigkeiten zwischen Signalen herzustellen.
+13. [ ] **Erweiterte Licht-Effekte:** Implementierung einer Bibliothek für Effekte wie Gaslaternen-Flackern, Leuchtstoffröhren-Zünden, Zufalls-Licht etc.
+14. [ ] **Sound-Unterstützung (Proof-of-Concept):** Ansteuerung eines kleinen Sound-Moduls (z.B. DFPlayer Mini), um signal-synchrone Sounds (z.B. Relais-Klicken, Glocke) abzuspielen.
+15. [ ] **Intelligentes Dimmen & Glühlampen-Simulation:** Überarbeitung der Dimm-Funktion, um das träge Auf- und Abglimmen von Glühlampen physikalisch korrekt zu simulieren.
+
+**Phase 4: Protokoll-Erweiterung & Zukunftssicherheit**
+16. [ ] **RailCom Channel 2 (BiDi-Daten):** Vollständige RailCom-Unterstützung für bidirektionalen Datenaustausch, z.B. für die Rückmeldung von Sensor-Daten.
+17. [ ] **mfx-Unterstützung (Anmeldung):** Implementierung der automatischen Anmeldung an mfx-Zentralen als Zubehör-Decoder.
+18. [ ] **CAN-Bus-Schnittstelle:** Hardware- und Software-Unterstützung für eine CAN-Bus-Schnittstelle zur Kommunikation mit anderen Baugruppen (z.B. für Gleisbesetztmelder).
+19. [ ] **Skripting-Engine (Lua/Python):** Integration einer leichtgewichtigen Skript-Engine, die es fortgeschrittenen Benutzern ermöglicht, eigene Logiken und Abläufe zu definieren.
+20. [ ] **JMRI-Decoder-Definitions-Dateien:** Erstellung und Pflege von JMRI-kompatiblen Decoder-Definitionsdateien, um die Konfiguration über JMRI zu vereinfachen.
```

Comment:
Verlinke wo immer möglich die zugehörigen Spezifikationen/Quellen in der Roadmap